### DISCUSSION TITLE
I got the token from the dashboard Haven [Apr 27]

### DISCUSSION THREAD

**Discussion by dstala** [nocodb/nocodb#5596] [Discord Threads]
> 🙋‍♂️ **saurav_mitra**

**Comment by o1lab** [comment]
> 👨‍💻 **pranavxc**  
#### I think you are using trying to pass auth token as api token    
####    
#### https://docs.nocodb.com/developer-resources/accessing-apis/#auth-token  
#  
> 👨‍💻 **pranavxc**  
#### https://docs.nocodb.com/setup-and-usages/team-and-auth#api-tokens-management  
#  
> 🙋‍♂️ **saurav_mitra**  
#### I've tried using xc-auth as well  
#  
> 🙋‍♂️ **saurav_mitra**  
#### I've tried both of these  
![image0](https://cdn.discordapp.com/attachments/1088160272415469610/1088163392650481796/image.png)  
![image1](https://cdn.discordapp.com/attachments/1088160272415469610/1088163392952487988/image.png)  
#  
> 🙋‍♂️ **saurav_mitra**  
#### <@682854952657616902> any ideas?  
#  
> 🙋‍♂️ **wkw**  
#### if u r using auth token, use `xc-auth`. if u r using api token, use `xc-token`  
#  
> 🙋‍♂️ **wkw**  
#### auth token can be expired - make sure u get it before sending the request  
#  
> 🙋‍♂️ **saurav_mitra**  
#### I tried both and neither worked. I got the auth token right before trying it.  
#  
> 🙋‍♂️ **saurav_mitra**  
#### Got the xc-auth to work today (not sure what the difference is from yesterday, it's the same script), but the xc-token still doesn't work 😢  
#  
> 👨‍💻 **pranavxc**  
#### can you share the following details   
####    
#### - Error you are getting when using auth-token    
#### - Are you defining xc-token header in request properly ?   
#### - How you generated the api token  
#  
> 🙋‍♂️ **saurav_mitra**  
#### Project -> Team & Settings -> API Tokens Management  
#  
> 🙋‍♂️ **saurav_mitra**  
#### `data: { msg: 'Unauthorized access' }`  
#  
> 👨‍💻 **pranavxc**  
#### And how the token is passing ? and any other header are you passing ?  
#  
> 🙋‍♂️ **saurav_mitra**  
#### no other headers  
#  

 [Autogenerated from discord](https://discord.com/channels/661905455894888490/1088160272415469610)  
- - - -  
#### Join [NocoDB's](http://www.nocodb.com/)  community  
[Discord](https://discord.gg/5RgZmkW) • [Twitter](https://twitter.com/nocodb) • [Reddit](https://www.reddit.com/r/NocoDB/) • [Docs](https://docs.nocodb.com/)
